src/wallet/wallet.cpp

```diff
@@ -1983,6 +1990,19 @@ CAmount CWallet::GetLegacyBalance(const isminefilter& filter, int minDepth, cons
     return balance;
 }
 
+CAmount CWallet::GetAvailableBalance(const CCoinControl* coinControl) const
+{
+    CAmount balance = 0;
```

Comment:
Not sure if a lock is required at this point. Do you want to protect `const CCoinControl* coinControl`?
`AvailableCoins()` does protect via `cs_main`/`cs_wallet`.